src/api/ApiCall.hh

```diff
@@ -11,21 +11,62 @@ namespace ra {
 namespace api {
 
 enum class ApiResult
-{    
+{
     None = 0,       // unspecified
     Success,        // call was successful
     Error,          // an error occurred
     Failed,         // call was unsuccessful, but didn't error
     Unsupported,    // call is not supported and was not made
+    Incomplete,     // call could not be made at this time
 };
 
 struct ApiRequestBase
 {
+protected:
     template<class TRequest, class TCallback>
     static void CallAsync(const TRequest& request, TCallback&& callback)
     {
         ra::services::ServiceLocator::GetMutable<ra::services::IThreadPool>().RunAsync(
-            [request, callback = std::move(callback)] { callback(request.Call()); });
+            [request, callback = std::move(callback)]{ callback(request.Call()); });
+    }
+
+    template<class TRequest, class TCallback>
+    static void CallAsyncWithRetry(const TRequest& request, TCallback&& callback)
+    {
+        ra::services::ServiceLocator::GetMutable<ra::services::IThreadPool>().RunAsync(
+            [request, callback = std::move(callback)]
+        {
+            DoAsyncWithRetry(std::move(request), std::move(callback), std::chrono::milliseconds(0));
```

Comment:
If you want to make this easier/more intuitive, you can try using the `chrono_literals`.
They have them for all duration helper types. They are all written the same as you would normally except microseconds uses the "us" suffix since the actual character for it is Unicode.

Not a big deal, just something to consider.
https://en.cppreference.com/w/cpp/chrono/duration